core/server/models/post.js

```diff
@@ -681,22 +682,49 @@ Post = ghostBookshelf.Model.extend({
             return unsafeAttrs.author_id && unsafeAttrs.author_id === context.user;
         }
 
-        if (baseUtils.actorIs(loadedPermissions.user, 'Contributor') && action === 'edit' && (isChanging('status') || postModel.get('status') !== 'draft')) {
-            // Contributors ONLY have permission to edit a post if they're not changing the status AND the post is a draft post
-            hasUserPermission = false;
-        } else if (baseUtils.actorIs(loadedPermissions.user, 'Contributor') && action === 'destroy' && postModel.get('status') !== 'draft') {
-            // Don't allow contributor to delete published posts
-            hasUserPermission = false;
-        } else if (baseUtils.actorIs(loadedPermissions.user, ['Author', 'Contributor']) && action === 'edit' && isChanging('author_id')) {
-            hasUserPermission = false;
-        } else if (baseUtils.actorIs(loadedPermissions.user, ['Author', 'Contributor']) && action === 'add') {
+        function isCurrentOwner() {
```

Comment:
These functions should probably be passed the arguments they use?

Reply:
hmm - are you talking about the `context`, `postModel`, and `unsafeAttr` properties? or are you talking about the strings like `draft`, `author_id`, etc.?